File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    arr=[words[0]]
    for i in range(1, len(words)):
        if words[i] in arr or not (len(words)>1 and words[i-1][-1] == words[i][0]):
            answer[0] = (i%n) + 1
            answer[1] = (i//n) + 1
            break
        arr.append(words[i])

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {0, 0};
        List<String> list = new ArrayList<>();
        
        for(int i=0;i<words.length; i++) {
            if (list.contains(words[i]) || (i > 0 && words[i].charAt(0) != words[i-1].charAt(words[i-1].length() - 1))) {
                answer[0] = (i%n) + 1;
                answer[1] = (i/n) + 1;
                return answer;
            }
            
            list.add(words[i]);
        }
        
        return new int[]{0, 0};
    }
}